test/integration/http2_upstream_integration_test.cc

```diff
@@ -345,23 +345,19 @@ name: router
   // As with ProtocolIntegrationTest.HittingEncoderFilterLimit use a filter
   // which buffers response data but in this case, make sure the sendLocalReply
   // is gRPC.
-  config_helper_.addFilter("{ name: envoy.filters.http.dynamo, typed_config: { \"@type\": "
+  config_helper_.addFilter("{ name: encoder-decoder-buffer-filter, typed_config: { \"@type\": "
                            "type.googleapis.com/google.protobuf.Empty } }");
   config_helper_.setBufferLimits(1024, 1024);
   initialize();
 
   // Send the request.
   codec_client_ = makeHttpConnection(lookupPort("http"));
-  auto encoder_decoder =
-      codec_client_->startRequest(Http::TestRequestHeaderMapImpl{{":method", "POST"},
-                                                                 {":path", "/test/long/url"},
-                                                                 {":scheme", "http"},
-                                                                 {":authority", "host"},
-                                                                 {"te", "trailers"}});
-  auto downstream_request = &encoder_decoder.first;
-  auto response = std::move(encoder_decoder.second);
-  Buffer::OwnedImpl data(R"({"TableName":"locations"})");
```

Comment:
This is a slightly different test (no request body now). Was there a reason to change this?

Reply:
It did not seem that the body was essential to the test but rather than to the dynamo filter expectation on body content. I had the test working with and without this change, so I have removed this code as it seemed to be dynamo specific. I can put it back if people feel it would be a safer change.

Reply:
My preference would be to keep the tests the same such that they buffer both request and response body?

Reply:
Done. I've changed the data string to make it clear the content does not matter.